core/sqf/src/seatrans/hbase-trx/src/main/java/org/apache/hadoop/hbase/coprocessor/transactional/TrxRegionEndpoint.java.tmpl

```diff
@@ -4015,6 +4015,25 @@ CoprocessorService, Coprocessor {
     if (LOG.isTraceEnabled()) LOG.trace("Trafodion Recovery: region " + m_regionDetails + " is STARTED.");
   }
 
+  public static void WALSync(WAL wal, long transactionId, long txid) throws IOException
+  {
+     try {
+       if (txid == 0)
+          wal.sync();
+       else
+          wal.sync(txid);
+     } catch (IOException wale) {
+        wale.fillInStackTrace();
+        LOG.error("commitRequest txId: " + transactionId + " HLog seq " + txid + " Caught IOException in HLOG sync ", wale );
+        try {
+           Thread.sleep(1000);          ///1000 milliseconds is one second.
```

Comment:
This is to stagger the sync operation assuming that it was causing the concurrentModificationException. But @shangshengtung came up with different change that took care of concurrentModificationException. He will be delivering his change in a separate PR soon